lib/mongoid/clients/factory.rb

```diff
@@ -57,6 +57,9 @@ def create_client(configuration)
         config = configuration.dup
         uri = config.delete(:uri)
         database = config.delete(:database)
+        if database.nil?
+          database = Mongo::URI.get(uri).database
+        end
```

Comment:
Is it worth condensing these four lines into one?

```suggestion
        database = config.delete(:database) || Mongo::URI.get(uri).database
```

Reply:
Good point, thank you!